streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -385,11 +385,16 @@ public void writeAsText(final String filePath,
             throw new TopologyException("filePath can't be an empty string");
         }
         String name = builder.newProcessorName(PRINTING_NAME);
+        PrintWriter printWriter = null;
         try {
-            PrintWriter printWriter = new PrintWriter(filePath, StandardCharsets.UTF_8.name());
+            printWriter = new PrintWriter(filePath, StandardCharsets.UTF_8.name());
             builder.internalTopologyBuilder.addProcessor(name, new KStreamPrint<>(new PrintForeachAction(printWriter, defaultKeyValueMapper, label)), this.name);
```

Comment:
@bbejeck @sebastiansemmle Was just thinking once more about this change. And it seem that the original code is actually correct. We pass in the `PrintWriter` into the builder and use it later on. If we close it here in `finally` we cannot use the `PrintWriter` later... And `PrintWriter` is closed correctly within `PrintForeachAction#close()`.

Reply:
You are right, indeed. It is supposed to be closed at a later point in time.

Reply:
Updated the PR to clean this up.